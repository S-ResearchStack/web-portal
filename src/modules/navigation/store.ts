import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';
import { isInsideTest } from 'src/common/utils/testing';

export enum Path {
  Root = '/',

  Registration = '/registration',
  SignIn = '/signin',
  AccountCreate = '/account/create',

  CreateStudy = '/create-study',

  OverviewSubject = '/overview/participants/:subjectNumber',
  StudyManagement = '/study-management',
  StudyManagementSubject = '/study-management/participants/:subjectNumber',
  StudyManagementEditSurvey = '/study-management/surveys/:surveyId/edit',
  StudyManagementSurveyResults = '/study-management/surveys/:surveyId/results',
  StudyManagementActivityResults = '/study-management/activities/:activityId/results',

  Overview = '/overview',
  SubjectManagement = '/subject-management',
  StudyData = '/study-data',
  StudySettings = '/study-settings',
  LabVisitManagement = '/lab-visit',

  Dashboard = '/dashboard',
  CreateChart = '/dashboard/:dashboardId/chart/create',
  EditChart = '/dashboard/:dashboardId/chart/:chartId/edit',

  TaskManagement = '/task-management',
  CreateSurvey = '/task-management/survey/create',
  CreateActivity = '/task-management/activity/:activityType/create',
  EditActivity = '/task-management/activity/:activityId/edit',

  EducationalManagement = '/educational-management',
  CreateEducational = '/educational-management/:educationType/create',
  EditEducational = '/educational-management/:educationId/edit',
}

export const makeHistory = () =>
  !isInsideTest
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.Dashboard,
  Path.TaskManagement,
  Path.SubjectManagement,
  Path.StudyData,
  Path.EducationalManagement,
  Path.LabVisitManagement,
  Path.StudySettings
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
