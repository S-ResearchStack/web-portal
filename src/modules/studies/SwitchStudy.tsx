import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useIntersection } from 'react-use';
import { push } from 'connected-react-router';
import _range from 'lodash/range';

import styled, { css } from 'styled-components';
import { useHistory } from 'react-router-dom';
import _chunk from 'lodash/chunk';

import { useAppDispatch, useAppSelector } from 'src/modules/store';
import {
  selectedStudySelector,
  selectStudy,
  studiesIsLoadingSelector,
  studiesSelector,
} from 'src/modules/studies/studies.slice';
import { Path } from 'src/modules/navigation/store';
import Button from 'src/common/components/Button';
import StudyAvatar from 'src/common/components/StudyAvatar';
import Plus from 'src/assets/icons/plus.svg';
import ArrowLeft from 'src/assets/icons/arrow_left_xxl.svg';
import ArrowRight from 'src/assets/icons/arrow_right_xxl.svg';
import { animation, colors, px, typography } from 'src/styles';
import SkeletonLoading, { SkeletonRect, SkeletonPath } from 'src/common/components/SkeletonLoading';
import ResultMessage from 'src/modules/auth/common/ResultMessage';
import EmptyStateImg from 'src/assets/illustrations/empty_state.svg';
import { SWITCH_STUDY_SEARCH_PARAM } from 'src/modules/main-layout/constants';
import { useTranslation } from '../localization/useTranslation';

const MAX_STUDIES_PER_SCREEN = 10;
const componentWidth = css`
  min-width: ${px(1120)};
  max-width: ${px(1120)};
`;

const Container = styled.div`
  width: 100%;
  height: 100%;
  background-color: ${colors.background};
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: auto;
`;

const ActionButton = styled(Button)<{ visible: boolean }>`
  margin-top: ${px(95)};
  height: ${({ visible }) => (visible ? px(80) : 0)};
`;

const Content = styled.div`
  ${componentWidth};
  color: ${colors.textPrimary};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  padding-top: ${px(34)};
`;

const Header = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: center;
  margin-bottom: ${px(68)};
  padding-top: ${px(5)};
`;

const CarouselWrapper = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

const StyledCarousel = styled.div`
  ${componentWidth};
  display: flex;
  overflow-x: hidden;
  margin-bottom: ${px(30)};
`;

const Inner = styled.div<{ index: number; total: number }>`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  white-space: nowrap;
  transform: ${({ index, total }) => `translateX(${(-100 * index) / total}%)`};
  transition: transform 0.3s ${animation.defaultTiming};
`;

const Studies = styled.div`
  ${componentWidth};
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  margin: auto;
  min-height: ${px(436)};
  row-gap: ${px(16)};
`;

const Title = styled.div`
  ${typography.headingMedium};
  margin-left: ${px(40)};
`;

const StudyContainer = styled.div<{ $selected: boolean | undefined }>`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: ${px(144)};
  height: ${px(227)};
  padding-top: ${({ $selected }) => ($selected ? 0 : px(4))};
  margin: 0 ${px(40)} ${px(8)};
`;

const StudyName = styled.div<{
  selected?: boolean;
  dimmed?: boolean;
  hovered?: boolean;
}>`
  ${({ selected, hovered }) =>
    !hovered && !selected ? typography.headingXMediumRegular : typography.headingXMedium};
  margin-top: ${px(26)};
  text-align: center;
  opacity: ${({ dimmed }) => dimmed && 0.75};
  max-width: ${px(144)};
  display: block;
  display: -webkit-box;
  height: fit-content;
  word-wrap: break-word;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: break-spaces;
`;

const StudyCounter = styled.div`
  ${typography.bodySmallRegular};
  color: ${colors.primary};
`;

const EmptyContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: ${px(12)};
`;

export type SwitchStudyProps = {
  onStudySelectionFinished?: () => void;
};

const SwitchStudy: React.FC<SwitchStudyProps> = ({ onStudySelectionFinished }) => {
  const isLoading = useAppSelector(studiesIsLoadingSelector);
  const studies = useAppSelector(studiesSelector);
  const selectedStudy = useAppSelector(selectedStudySelector);
  const dispatch = useAppDispatch();
  const [hoveredItemIdx, setHoveredItemIdx] = useState<number | null>(null);
  const [slideIndex, setSlideIndex] = useState<number>(0);
  const [selectedId, setSelectedId] = useState<string>('');
  const history = useHistory();
  const ref = useRef<HTMLDivElement>(null);
  const isVisible = !!useIntersection(ref, {})?.isIntersecting;

  useEffect(() => {
    if (!isVisible) {
      setSelectedId('');
    }
  }, [isVisible]);

  const studyPages = useMemo(
    () =>
      _chunk(
        studies.map((s) => ({ ...s, selected: selectedId ? s.id === selectedId : undefined })),
        MAX_STUDIES_PER_SCREEN
      ).map((ss, idx) => ({
        id: idx,
        studies: ss,
      })),
    [selectedId, studies]
  );

  const handleStudySelected = (id: string) => {
    if (id !== selectedStudy?.id) {
      history.replace(Path.Overview);
    }

    if (history.location.search.includes(SWITCH_STUDY_SEARCH_PARAM)) {
      history.replace(history.location.pathname);
    }

    dispatch(selectStudy(id));
    onStudySelectionFinished?.();
  };

  const createStudy = () => {
    dispatch(push(Path.CreateStudy));
  };

  const isHovered = Number.isFinite(hoveredItemIdx);

  const Counter = useCallback(
    (index: number) => {
      let counter = <>&nbsp;</>;
      if (studyPages.length > 1) {
        counter = (
          <>
            {Math.abs(index) * MAX_STUDIES_PER_SCREEN + studyPages[index].studies.length} of{' '}
            {studies.length}
          </>
        );
      }
      return counter;
    },
    [studyPages, studies.length]
  );

  const loadingRow = useMemo(
    () =>
      _range(0, MAX_STUDIES_PER_SCREEN).map((idx) => (
        <StudyContainer key={idx} $selected={false}>
          <SkeletonLoading>
            <SkeletonPath
              x="0"
              y="0"
              d="M141.23 100.118C135.335 127.551 120.189 144 72.2835 144C23.5103 144 9.07086 127.811 2.85704 100.119C1.18261 92.6562 0.0788032 81.4557 0 72.7101C0 63.5616 1.11855 51.6301 2.85704 43.8817C9.07086 16.1894 23.5103 0 72 0C120.189 0 135.335 16.4487 141.23 43.8817C143.042 52.3163 144 62.0734 144 72.426C143.941 82.6929 142.984 91.9541 141.23 100.118Z"
            />
            <SkeletonRect x="0" y="168" rx={0} ry={0} width={144} height={27} />
          </SkeletonLoading>
        </StudyContainer>
      )),
    []
  );
  const {t} = useTranslation()
  return (
    <Container>
      <ActionButton
        icon={<ArrowLeft />}
        fill="text"
        width={80}
        onClick={() => setSlideIndex(slideIndex - 1)}
        disabled={slideIndex === 0}
        visible={studyPages.length > 1}
        aria-label="Previous Study"
      />
      <Content>
        <Header>
          <Title>{t("TITLE_STUDY_COLLECTION")}</Title>
          <Button fill="text" width={246} icon={<Plus />} onClick={createStudy} rippleOff>
            {t("TITLE_CREATE_STUDY")}
          </Button>
        </Header>
        {/* eslint-disable-next-line no-nested-ternary */}
        {studyPages.length ? (
          <CarouselWrapper ref={ref}>
            <StyledCarousel>
              <Inner index={slideIndex} total={studyPages.length}>
                {studyPages.map((page) => (
                  <Studies key={page.id}>
                    {page.studies.map(({ id, name, color, selected }, idx) => {
                      const isItemHovered = hoveredItemIdx === idx;
                      return (
                        <StudyContainer
                          $selected={selected}
                          onMouseUp={() => handleStudySelected(id)}
                          onMouseDown={() => setSelectedId(id)}
                          key={id}
                        >
                          <StudyAvatar
                            color={color}
                            size="xxl"
                            $selected={selected}
                            onMouseEnter={() => setHoveredItemIdx(idx)}
                            onMouseLeave={() => setHoveredItemIdx(null)}
                          />
                          <StudyName
                            selected={selected}
                            hovered={isItemHovered}
                            dimmed={selected && isHovered && !isItemHovered}
                          >
                            {name}
                          </StudyName>
                        </StudyContainer>
                      );
                    })}
                  </Studies>
                ))}
              </Inner>
            </StyledCarousel>
          </CarouselWrapper>
        ) : !isLoading ? (
          <EmptyContainer>
            <ResultMessage
              moreSpace
              picture={<EmptyStateImg />}
              title="No studies yet"
              description="Access studies you have been invited to here."
            />
          </EmptyContainer>
        ) : (
          <CarouselWrapper>
            <StyledCarousel>
              <Studies>{loadingRow}</Studies>
            </StyledCarousel>
          </CarouselWrapper>
        )}
        <StudyCounter>{Counter(slideIndex)}</StudyCounter>
      </Content>
      <ActionButton
        icon={<ArrowRight />}
        fill="text"
        width={80}
        onClick={() => setSlideIndex(slideIndex + 1)}
        disabled={slideIndex === studyPages.length - 1}
        visible={studyPages.length > 1}
        aria-label="Next Study"
      />
    </Container>
  );
};

export default SwitchStudy;
