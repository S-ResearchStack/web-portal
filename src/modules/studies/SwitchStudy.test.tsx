import { screen, render } from '@testing-library/react';
import _range from 'lodash/range';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import React from 'react';
import { ConnectedRouter } from 'connected-react-router';
import { history } from 'src/modules/navigation/store';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import userEvent from '@testing-library/user-event';
import { theme } from 'src/styles';
import { createTestStore } from 'src/modules/store/testing';
import SwitchStudy from './SwitchStudy';

describe('SwitchStudy', () => {
  it('should render', async () => {
    const store = createTestStore({
      studies: {
        studies: _range(0, 20).map((idx) => ({
          id: `study-${idx}`,
          name: `Study ${idx}`,
          color: 'secondarySkyBlue',
          createdAt: 1652648400000,
        })),
        selectedStudyId: 'test',
        isLoading: false,
      },
    });

    const { baseElement } = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <SwitchStudy />
          </ConnectedRouter>
        </Provider>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    await userEvent.click(screen.getByLabelText('Next Study'));
    await userEvent.click(screen.getByLabelText('Previous Study'));
    await userEvent.click(screen.getByText('Create Study'));

    await userEvent.hover(screen.getAllByTestId('avatar-icon')[0]);
    await userEvent.unhover(screen.getAllByTestId('avatar-icon')[0]);

    await userEvent.click(screen.getAllByTestId('avatar-icon')[0]);
  });

  it('[NEGATIVE] should render with empty store', async () => {
    const store = createTestStore({});

    const { baseElement } = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <SwitchStudy />
          </ConnectedRouter>
        </Provider>
      </ThemeProvider>
    );

    expect(baseElement).toBeInTheDocument();
  });
});
