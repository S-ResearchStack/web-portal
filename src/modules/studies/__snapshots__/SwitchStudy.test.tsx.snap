// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchStudy should render 1`] = `
.c4 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c1 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 5rem;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: transparent;
}

.c1 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c1 > div svg {
  margin-right: 0.25rem;
}

.c1 > div {
  color: #4475E3;
}

.c1 > div svg {
  fill: #4475E3;
}

.c1:hover:enabled > div {
  color: #2F65E0;
}

.c1:hover:enabled > div svg {
  fill: #2F65E0;
}

.c1:hover:enabled:active > div {
  color: #2057D5;
}

.c1:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c1:disabled > div {
  color: #B3C6F1;
}

.c1:disabled > div svg {
  fill: #B3C6F1;
}

.c1:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c1:focus-visible > div {
  color: #4475E3;
}

.c1:focus-visible > div svg {
  fill: #4475E3;
}

.c8 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 15.375rem;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: transparent;
}

.c8:hover {
  cursor: pointer;
}

.c8 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c8 > div svg {
  margin-right: 0.25rem;
}

.c8 > div {
  color: #4475E3;
}

.c8 > div svg {
  fill: #4475E3;
}

.c8:hover:enabled > div {
  color: #2F65E0;
}

.c8:hover:enabled > div svg {
  fill: #2F65E0;
}

.c8:hover:enabled:active > div {
  color: #2057D5;
}

.c8:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c8:disabled > div {
  color: #B3C6F1;
}

.c8:disabled > div svg {
  fill: #B3C6F1;
}

.c8:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c8:focus-visible > div {
  color: #4475E3;
}

.c8:focus-visible > div svg {
  fill: #4475E3;
}

.c17 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 5rem;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: transparent;
}

.c17:hover {
  cursor: pointer;
}

.c17 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c17 > div svg {
  margin-right: 0.25rem;
}

.c17 > div {
  color: #4475E3;
}

.c17 > div svg {
  fill: #4475E3;
}

.c17:hover:enabled > div {
  color: #2F65E0;
}

.c17:hover:enabled > div svg {
  fill: #2F65E0;
}

.c17:hover:enabled:active > div {
  color: #2057D5;
}

.c17:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c17:disabled > div {
  color: #B3C6F1;
}

.c17:disabled > div svg {
  fill: #B3C6F1;
}

.c17:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c17:focus-visible > div {
  color: #4475E3;
}

.c17:focus-visible > div svg {
  fill: #4475E3;
}

.c3 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c14 {
  border-radius: 4.125rem;
  height: 8.875rem;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  position: relative;
  background-color: transparent !important;
  opacity: 1;
}

.c14:hover {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
}

.c14:hover:active {
  opacity: 1;
  box-shadow: 0 0.9375rem 1.25rem rgba(180,180,180,0.3);
  cursor: pointer;
  -webkit-filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
  filter: drop-shadow(0 0.9375rem 1.25rem rgba(180,180,180,0.3));
}

.c0 {
  width: 100%;
  height: 100%;
  background-color: #F7F8FA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: auto;
}

.c2 {
  margin-top: 5.9375rem;
  height: 5rem;
}

.c5 {
  min-width: 70rem;
  max-width: 70rem;
  color: #474747;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 2.125rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 4.25rem;
  padding-top: 0.3125rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  min-width: 70rem;
  max-width: 70rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: hidden;
  margin-bottom: 1.875rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.45,0.05,0,1);
  -webkit-transition: transform 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: transform 0.3s cubic-bezier(0.45,0.05,0,1);
}

.c12 {
  min-width: 70rem;
  max-width: 70rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
  min-height: 27.25rem;
  row-gap: 1rem;
}

.c7 {
  font-family: Inter,sans-serif;
  font-size: 1.5rem;
  line-height: 2.0125rem;
  font-weight: 600;
  font-style: normal;
  margin-left: 2.5rem;
}

.c13 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 9rem;
  height: 14.1875rem;
  padding-top: 0.25rem;
  margin: 0 2.5rem 0.5rem;
}

.c15 {
  font-family: Inter,sans-serif;
  font-size: 1.125rem;
  line-height: 1.6875rem;
  font-weight: 400;
  font-style: normal;
  margin-top: 1.625rem;
  text-align: center;
  max-width: 9rem;
  display: block;
  display: -webkit-box;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  word-wrap: break-word;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: break-spaces;
}

.c16 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #4475E3;
}

<body>
  <div>
    <div
      class="c0"
    >
      <button
        aria-label="Previous Study"
        class="c1 c2"
        disabled=""
        fill="text"
        style="overflow: hidden; position: relative;"
        width="80"
      >
        <div
          class="c3"
          data-testid="content"
        >
          <svg />
        </div>
        <div
          class="c4"
          color="primaryLightPressed"
          opacity="1,1"
          scale="0"
          style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
        />
      </button>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            Study Collection
          </div>
          <button
            class="c8"
            fill="text"
            width="246"
          >
            <div
              class="c3"
              data-testid="content"
            >
              <svg />
              Create Study
            </div>
          </button>
        </div>
        <div
          class="c9"
        >
          <div
            class="c10"
          >
            <div
              class="c11"
            >
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 0
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 1
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 2
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 3
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 4
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 5
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 6
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 7
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 8
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 9
                  </div>
                </div>
              </div>
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 10
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 11
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 12
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 13
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 14
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 15
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 16
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 17
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 18
                  </div>
                </div>
                <div
                  class="c13"
                >
                  <div
                    class="c14"
                    color="secondarySkyBlue"
                  >
                    <svg
                      data-testid="avatar-icon"
                      fill="#00B0D7"
                    />
                  </div>
                  <div
                    class="c15"
                  >
                    Study 19
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c16"
        >
          10
           of
           
          20
        </div>
      </div>
      <button
        aria-label="Next Study"
        class="c17 c2"
        fill="text"
        style="overflow: hidden; position: relative;"
        width="80"
      >
        <div
          class="c3"
          data-testid="content"
        >
          <svg />
        </div>
        <div
          class="c4"
          color="primaryLightPressed"
          opacity="1,1"
          scale="0"
          style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
        />
      </button>
    </div>
  </div>
</body>
`;
